bigframes/core/nodes.py

```diff
@@ -409,6 +409,49 @@ def transform_children(
         return self
 
 
+@dataclass(frozen=True)
+class CachedTableNode(BigFrameNode):
+    # The original BFET subtree
+    # note: this isn't a "child" node.
+    original_node: BigFrameNode = field()
+    # reference to cached materialization
+    project_id: str = field()
+    dataset_id: str = field()
+    table_id: str = field()
+    physical_schema: Tuple[bq.SchemaField, ...] = field()
+
+    ordering: orderings.ExpressionOrdering = field()
+
+    @property
+    def session(self):
+        return self.original_node.session
+
+    def __hash__(self):
+        return self._node_hash
```

Comment:
Where is this initialized?

Thoughts on calling `hash(self.original_node)`, instead?

Reply:
The _node_hash is a property defined on the base class that introspects all the dataclass fields to generate a hash. Generating an equivalent hash to `original_node` is an interesting idea, but I think for the most part, we want hash to represent structural equivalence rather than going for some sort of imperfect semantic equivalence.